File: sdk-dotnet/src/HealthFlow.Hfcx.Sdk/Recipient/Layer.cs
// Copyright (c) HealthFlow Medical HCX. Licensed under Apache 2.0.

namespace HealthFlow.Hfcx.Sdk.Recipient;

/// <summary>
/// Pipeline layers in <see cref="RecipientHandler"/>, each independently
/// toggleable. Cross-SDK invariant with Java's <c>Layer</c> and Python's
/// <c>Layer</c> enum: 4 values in this exact order.
/// </summary>
public enum Layer
{
    /// <summary>Bearer-token validation.</summary>
    Bearer,

    /// <summary>Five protocol-header presence + format checks.</summary>
    Headers,

    /// <summary>FHIR-Bundle Egyptian-IG profile validation.</summary>
    Fhir,

    /// <summary>Egyptian field-value walks (NID / phone / IBAN).</summary>
    Egyptian,
}

File: sdk-javascript/src/recipient/Layer.ts
// Copyright (c) HealthFlow Medical HCX. Licensed under Apache 2.0.

/**
 * Pipeline layers in {@link RecipientHandler}, each independently
 * toggleable. Cross-SDK invariant with Java's `Layer`, Python's
 * `Layer`, and .NET's `Layer`: 4 values in this exact order.
 */
export type Layer = 'BEARER' | 'HEADERS' | 'FHIR' | 'EGYPTIAN';

export const Layer = {
  BEARER: 'BEARER',
  HEADERS: 'HEADERS',
  FHIR: 'FHIR',
  EGYPTIAN: 'EGYPTIAN',
} as const satisfies Record<Layer, Layer>;

export const ALL_LAYERS: readonly Layer[] = [
  Layer.BEARER,
  Layer.HEADERS,
  Layer.FHIR,
  Layer.EGYPTIAN,
];
